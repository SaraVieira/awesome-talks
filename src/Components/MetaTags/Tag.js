import React from 'react'
import { Helmet } from 'react-helmet'
import humanize from '../../Utils/strings'

export default ({ category }) => (
    <Helmet>
        <title>Awesome Talks - {category}</title>
        <meta
            name="description"
            content="Amazing Tech Talks curated by the community ❤️"
        />
        <meta
            name="twitter:title"
            content={`Awesome Talks - ${humanize(category)}`}
        />
        <meta
            name="twitter:description"
            content="Amazing Tech Talks curated by the community ❤️"
        />
        <meta name="twitter:image" content="https://file-iloqdynwox.now.sh/" />
        <meta name="twitter:image:alt" content="awesome talks" />
    </Helmet>
)
