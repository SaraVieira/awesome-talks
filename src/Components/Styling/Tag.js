import styled from 'styled-components'
import { Link } from 'react-router-dom'
import remcalc from 'remcalc'

export default styled(Link)`
    font-family: 'Space Mono', monospace;
    border: none;
    color: ${props => props.theme.secondary};
    margin-top: ${remcalc(-10)};
    padding: 0;
    margin-bottom: ${remcalc(10)};
    margin-right: ${remcalc(10)};

    &:hover {
        opacity: 1;
        color: ${props => props.theme.main};
    }
    &:after {
        display: none;
    }
`
