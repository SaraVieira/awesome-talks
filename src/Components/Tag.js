import React from 'react'
import Item from './../Components/Styling/Item'

const makeTag = name => name.replace(/\s+/g, '-').toLowerCase()
const makeLink = name => `/category/${makeTag(name)}`

export default ({ name, id, _videosMeta: { count } }) => (
    <Item className="no-hover" to={makeLink(name)}>
        <span>#{makeTag(name)}</span>
        <span className="count">
            {count}
            {count > 1 ? ' Videos' : ' Video'}
        </span>
    </Item>
)
