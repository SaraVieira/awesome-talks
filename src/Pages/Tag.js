import React, { Fragment } from 'react'
import Header from './../Components/Header'
import { Col, Row, Grid } from 'react-styled-flexboxgrid'
import Query from './../Components/Query'
import Video from './../Components/Video'
import TAG_VIDEOS from '../Queries/TAG_VIDEOS'
import humanize from '../Utils/strings'
import Nav from './../Components/Nav'
import { Helmet } from 'react-helmet'
import CookieBanner from './../Components/CookieBanner'
import Error404 from './../Components/Errors/Error404'

export default ({
    match: {
        params: { category }
    }
}) => (
    <Grid>
        <Helmet>
            <title>Awesome Talks - {category}</title>
            <meta
                name="description"
                content="Amazing Tech Talks curated by the community ❤️"
            />
            <meta
                name="twitter:title"
                content={`Awesome Talks - ${humanize(category)}`}
            />
            <meta
                name="twitter:description"
                content="Amazing Tech Talks curated by the community ❤️"
            />
            <meta
                name="twitter:image"
                content="https://file-iloqdynwox.now.sh/"
            />
            <meta name="twitter:image:alt" content="awesome talks" />
        </Helmet>
        <div role="banner">
            <Nav />
        </div>
        <Query query={TAG_VIDEOS} variables={{ name: humanize(category) }}>
            {({ data: { allTagses } }) => (
                <Fragment>
                    {!allTagses.length || !allTagses[0].videos.length ? (
                        <Error404 />
                    ) : (
                        <Fragment>
                            <Header
                                title={`#${humanize(category)}`}
                                noSearch
                                code
                            />
                            <Row>
                                <Col xs={12}>
                                    <main>
                                        <Row>
                                            {allTagses[0].videos.map(v => (
                                                <Video key={v.id} talk={v} />
                                            ))}
                                        </Row>
                                    </main>
                                </Col>
                            </Row>
                        </Fragment>
                    )}
                </Fragment>
            )}
        </Query>
        <CookieBanner />
    </Grid>
)
