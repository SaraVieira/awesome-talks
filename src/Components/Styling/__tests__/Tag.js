import React from 'react'
import Shallow from '../../../Tests/shallow'
import { MemoryRouter } from 'react-router-dom'
import Tag from '../Tag'

test('Renders Tag', () => {
    const Wrapper = Shallow(
        <MemoryRouter>
            <Tag />
        </MemoryRouter>
    )
    expect(Wrapper).toMatchSnapshot()
})
