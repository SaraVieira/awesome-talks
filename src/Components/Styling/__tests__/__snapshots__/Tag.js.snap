// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Tag 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "jv0jkk",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "jv0jkk",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Styled(Link) />
</Router>
`;
